clippy_lints/src/ignored_unit_patterns.rs

```diff
@@ -37,6 +38,17 @@ declare_lint_pass!(IgnoredUnitPatterns => [IGNORED_UNIT_PATTERNS]);
 
 impl<'tcx> LateLintPass<'tcx> for IgnoredUnitPatterns {
     fn check_pat(&mut self, cx: &LateContext<'tcx>, pat: &'tcx hir::Pat<'tcx>) {
+        match get_parent_node(cx.tcx, pat.hir_id) {
```

Comment:
To be honest, I would even go further rather than use useless defensive programming by returning in the `else`. Would you concur that there should not exist any situation in which a pattern will not have a parent node (and moreover reach this method)?  I could use `.unwrap()` here instead of a `let/else`.

Reply:
Done.